app/services/processors/mappings.rb

```diff
@@ -23,5 +23,25 @@ def self.create(specification, user)
       )
       m
     end
+
+    ###
+    # @description: Creates the terms for the mapping.
+    # @param [Mapping] mapping: The mapping for which the terms are going to be created.
+    # @params [Array] terms: The list of terms as an array of objects. These are the
+    #   specification terms, selected to be mapped against the spine.
+    #
+    # @return [Mapping]
+    ###
+    def self.create_terms(mapping, terms)
+      terms.each do |term|
```

Comment:
What do you think about wrapping this loop into a transaction? So it's all or nothing, if an error occurs.